src/App.js

```diff
@@ -1,22 +1,33 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
-function App() {
-    const [todoList, setTodoList] = useState([]);
+function useSemiPersistentState(key, initialState) {
+    const savedState = JSON.parse(localStorage.getItem(key)) || initialState;
+    const [state, setState] = useState(savedState);
```

Comment:
Nice use of generic variable names to increase the modularity of this component!